pkg/backend/diy/backend.go

```diff
@@ -371,17 +371,21 @@ func (b *diyBackend) Upgrade(ctx context.Context, opts *UpgradeOptions) error {
 		return olds[i].Name().String() < olds[j].Name().String()
 	})
 
-	// There's no limit to the number of stacks we need to upgrade.
+	// Get the parallel value from environment variable or use a default value
+	parallel := b.Env.GetInt(env.DIYBackendParallel)
+	if parallel <= 0 {
+		parallel = 10 // Default to 10 parallel operations if not specified
```

Comment:
Maybe add a comment that 10 is just a sensible guess at a good number and hasn't been tuned.